### PR TITLE
Multiapp Workload: Workload to test how responsive a device is when…

### PR DESCRIPTION
**Pull request by mcgeagh**
… context switching between appplication tasks. It combines workflows from googlephotos, gmail and skype.

Added tagName to give unique logger name for multiple share attempts. Turn off markers_enabled for subclasses.

This is dependent on #261 #257 #252 being merged.
This is also dependent on workload-automation-assets#3 being merged.

### CODE REVIEW

File: wlauto/workloads/multiapp/__init__.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    12. The second image is posted to recipient
+    '''
+
+    parameters = [
+        Parameter('test_image', kind=str, default='uxperf_1600x1200.jpg',
+                  description='''
+                  An image to be copied onto the device that will be shared
+                  across multiple apps
+                  '''),
+        Parameter('recipient', kind=str, mandatory=True,
```
Comment (setrofim) [inline wlauto/workloads/multiapp/__init__.py]: This should not be mandatory. There various ways of specifying a default here. E.g. create a dedicated test account in gmail, or use mailinator.com

File: wlauto/workloads/multiapp/__init__.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                  description='''
+                  The email address of the recipient.  Setting a void address
+                  will stop any mesage failures clogging up your device inbox
+                  '''),
+        Parameter('login_name', kind=str, mandatory=True,
```
Comment (setrofim) [inline wlauto/workloads/multiapp/__init__.py]: It's not clear what login names and passwords need to be provided here -- this is not covered in the setup instructions.

File: wlauto/workloads/multiapp/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        // once more from googlephotos
+        pressBack();
+        pressBack();
+        sendToSkype(contactName);
+
+        unsetScreenOrientation();
+    }
+
+    private void setupGooglePhotos() throws Exception {
+        sleep(5); // Pause while splash screen loads
```
Comment (setrofim) [inline wlauto/workloads/multiapp/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java]: Is a fixed delay necessary here? Can you not wait for the splash screen UI  element to first appear and then disappear here?

File: wlauto/workloads/multiapp/__init__.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+import os
+
+from wlauto import AndroidUxPerfWorkload, Parameter
+from wlauto.exceptions import ValidationError
+
+
+class Multiapp(AndroidUxPerfWorkload):
+
+    name = 'multiapp'
```
Comment (mcgeagh) [inline wlauto/workloads/multiapp/__init__.py]: Renamed to appshare
Dependent on workload-automation-assets#5

File: wlauto/workloads/multiapp/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java (hunk 3 above)
Comment (mcgeagh) [inline wlauto/workloads/multiapp/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java]: Was legacy code. Tested without on latest changes to GooglePhotos. Passes across the 5 devices we have.

**Comment by mcgeagh** [comment]
Works on S7 and Mate8.
Fails on S5 because Google Photos crashes on this device.
Crash is not related to wa or this workload.

**Pull request merged**